src/relativedeltafield/fields.py

```diff
@@ -47,7 +76,11 @@ def get_db_prep_value(self, value, connection, prepared=False):
         if value is None:
             return value
         else:
-            return format_relativedelta(self.to_python(value))
+            eng = settings.DATABASES[self.model._meta.default_manager.db]['ENGINE']
```

Comment:
I tested this PR in a real-world project, and the code crashed on this line:

```
AttributeError: 'RelativeDeltaField' object has no attribute 'model'
```

This was in code which used `RelativeDeltaField` as an `output_field` in a complex query, which is why there is no model associated with the field.

Essentially, I annotate a queryset with an expression which subtracts a month from a field. I don't remember why but I had to cast it to a DurationField for some reason. Something along the lines of:

```
one_month = Cast(Value(relativedelta(months=1), output_field=RelativeDeltaField()), DurationField())
Foo.objects.annotate(month_earlier=F('something')-one_month)
```

Could you come up with a test which triggers this error and then fix it?

Reply:
I have no idea what exactly to do :-)
If you post here your model and a snippet you'd like to to work (or even the test) I can work with it and see what I can do.

Reply:
That's thousands of lines of overly complicated hairy and proprietary project-specific code. I doubt that would be useful to you for debugging this :)

Have you tried something like the code snippet I did post?

Reply:
I'm not asking the 1000s lines... just an example stripped down to the bare minimum... As so far I never used Cast or Value any context is better than nothing.
Did you code work before?
I would expect the field's model attribute to be set by the contribute_to_class which I did not override.

Reply:
is "something" a DateField ?

Reply:
Found the issue and applied a workaround. However the whole multi-db concept works only up to a certain extent (so far). I added a test I believe should check your scenario but it's commented out since it's not working yet.
Apparently Django needs the get_db_prep_value when building the query. However my multi-db implementation works only if the field knows on which db he's running. In aggregations field is not bound to a model Hence the reason of the missing attribute.
Nothing I can do there I'm afraid: that I know of there is no way I can get to the DB engine if the DB field is not bound